app/renderer/util.js

```diff
@@ -114,69 +121,122 @@ function isXpathUnique(xpath, doc, domNode) {
   return [true];
 }
 
+/**
+ * Given an xml doc and a current dom node, try to find a robust xpath selector qualified by
+ * key attributes, which is unique in the document (or unique plus index).
+ *
+ * @param {string} xpath
+ * @param {DOMDocument} doc
+ * @param {Array<string>|Array<[string, string]>} attrs - a list of attributes to consider, or
+ * a list of pairs of attributes to consider in conjunction
+ *
+ * @returns {[string|undefined, boolean|undefined]} tuple consisting of (1) the xpath selector discovered, and (2)
+ * whether this selector is absolutely unique in the document (true) or qualified by index (false)
+ */
+function getUniqueXPath(doc, domNode, attrs) {
+  let uniqueXpath, semiUniqueXpath;
+  const tagForXpath = domNode.tagName || '*';
+  const isPairs = attrs.length > 0 && _.isArray(attrs[0]);
+
+  for (const attrName of attrs) {
+    let xpath;
+    if (isPairs) {
+      const [attr1Name, attr2Name] = attrName;
+      const [attr1Value, attr2Value] = attrName.map(domNode.getAttribute.bind(domNode));
+      if (!attr1Value || !attr2Value) {
+        continue;
+      }
+      xpath = `//${tagForXpath}[@${attr1Name}="${attr1Value}" and @${attr2Name}="${attr2Value}"]`;
+    } else {
+      const attrValue = domNode.getAttribute(attrName);
+      if (!attrValue) {
+        continue;
+      }
+      xpath = `//${tagForXpath}[@${attrName}="${attrValue}"]`;
+    }
+    const [isUnique, indexIfNotUnique] = isXpathUnique(xpath, doc, domNode);
```

Comment:
a common agreement for function starting with `is` would be to return a boolean value. How about renaming this one to determineXpathUniqueness?